<Type Name="NullReferenceException" FullName="System.NullReferenceException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="858a623b57c3198fdbd963ea2c71f46d4bab951e" />
    <Meta Name="ms.sourcegitcommit" Value="6a0b904069161bbaec4ffd02aa7d9cf38c61e72e" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="es-ES" />
    <Meta Name="ms.lasthandoff" Value="06/24/2018" />
    <Meta Name="ms.locfileid" Value="36429290" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class NullReferenceException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit NullReferenceException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.NullReferenceException" />
  <TypeSignature Language="VB.NET" Value="Public Class NullReferenceException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class NullReferenceException : SystemException" />
  <TypeSignature Language="F#" Value="type NullReferenceException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
    <AssemblyVersion>4.0.20.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="1f0fd-101">Excepción que se produce cuando se intenta desreferenciar un objeto null.</span>
      <span class="sxs-lookup">
        <span data-stu-id="1f0fd-101">The exception that is thrown when there is an attempt to dereference a null object reference.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1f0fd-102">A <xref:System.NullReferenceException> excepción se produce cuando intenta tener acceso a un miembro de un tipo cuyo valor es `null`.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-102">A <xref:System.NullReferenceException> exception is thrown when you try to access a member on a type whose value is `null`.</span></span> <span data-ttu-id="1f0fd-103">Un <xref:System.NullReferenceException> excepción normalmente refleja el error de desarrollador y se produce en los escenarios siguientes:</span><span class="sxs-lookup"><span data-stu-id="1f0fd-103">A <xref:System.NullReferenceException> exception typically reflects developer error and is thrown in the following scenarios:</span></span>  
  
-   <span data-ttu-id="1f0fd-104">Ha olvidado crear instancias de un tipo de referencia.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-104">You've forgotten to instantiate a reference type.</span></span> <span data-ttu-id="1f0fd-105">En el ejemplo siguiente, `names` se declara, pero nunca se crean instancias:</span><span class="sxs-lookup"><span data-stu-id="1f0fd-105">In the following example, `names` is declared but never instantiated:</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/example1.cs#1)]
     [!code-vb[System.NullReferenceException.Class#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/example1.vb#1)]  
  
     <span data-ttu-id="1f0fd-106">Algunos compiladores emiten una advertencia cuando compila este código.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-106">Some compilers issue a warning when they compile this code.</span></span> <span data-ttu-id="1f0fd-107">Otros emiten un error y se produce un error en la compilación.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-107">Others issue an error, and the compilation fails.</span></span> <span data-ttu-id="1f0fd-108">Para resolver este problema, una instancia del objeto para que su valor ya no es `null`.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-108">To address this problem, instantiate the object so that its value is no longer `null`.</span></span> <span data-ttu-id="1f0fd-109">El siguiente ejemplo realiza llamando al constructor de clase de un tipo.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-109">The following example does this by calling a type's class constructor.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/example1a.cs#2)]
     [!code-vb[System.NullReferenceException.Class#2](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/example1a.vb#2)]  
  
-   <span data-ttu-id="1f0fd-110">Ha olvidado a una matriz de dimensión antes de inicializarlo.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-110">You've forgotten to dimension an array before initializing it.</span></span> <span data-ttu-id="1f0fd-111">En el ejemplo siguiente, `values` se declara como una matriz de enteros, pero nunca se especifica el número de elementos que contiene.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-111">In the following example, `values` is declared to be an integer array, but the number of elements that it contains is never specified.</span></span> <span data-ttu-id="1f0fd-112">El intento de inicializar sus valores, por tanto, se produce un <xref:System.NullReferenceException> excepción.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-112">The attempt to initialize its values therefore thrown a  <xref:System.NullReferenceException> exception.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#10](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Array3.cs#10)]
     [!code-vb[System.NullReferenceException.Class#10](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Array3.vb#10)]  
  
     <span data-ttu-id="1f0fd-113">Puede eliminar la excepción declarando el número de elementos de la matriz antes de inicializar, igual que en el ejemplo siguiente.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-113">You can eliminate the exception by declaring the number of elements in the array before initializing it, as the following example does.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#11](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Array4.cs#11)]
     [!code-vb[System.NullReferenceException.Class#11](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Array4.vb#11)]  
  
     <span data-ttu-id="1f0fd-114">Para obtener más información sobre cómo declarar e inicializar matrices, vea [matrices](http://msdn.microsoft.com/library/bb79bdde-e570-4c30-adb0-1dd5759ae041) y [matrices](http://msdn.microsoft.com/library/dbf29737-b589-4443-bee6-a27588d9c67e).</span><span class="sxs-lookup"><span data-stu-id="1f0fd-114">For more information on declaring and initializing arrays, see [Arrays](http://msdn.microsoft.com/library/bb79bdde-e570-4c30-adb0-1dd5759ae041) and [Arrays](http://msdn.microsoft.com/library/dbf29737-b589-4443-bee6-a27588d9c67e).</span></span>  
  
-   <span data-ttu-id="1f0fd-115">Obtendrá un **null** valor devuelto de un método y, a continuación, llamar a un método en el tipo de valor devuelto.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-115">You get a **null** return value from a method, and then call a method on the returned type.</span></span> <span data-ttu-id="1f0fd-116">A veces, esto es el resultado de un error de documentación; se produce un error en la documentación de tener en cuenta que puede devolver una llamada al método `null`.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-116">This sometimes is the result of a documentation error; the documentation fails to note that a method call can return `null`.</span></span> <span data-ttu-id="1f0fd-117">En otros casos, el código de forma errónea se da por supuesto que el método siempre devolverá no**null** valor.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-117">In other cases, your code erroneously assumes that the method will always return a non-**null** value.</span></span>  
  
     <span data-ttu-id="1f0fd-118">El código en el ejemplo siguiente se da por supuesto que el <xref:System.Array.Find%2A?displayProperty=nameWithType> método siempre devuelve `Person` cuyos `FirstName` campo coincide con una cadena de búsqueda.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-118">The code in the following example assumes that the <xref:System.Array.Find%2A?displayProperty=nameWithType> method always returns `Person` object whose `FirstName` field matches a search string.</span></span> <span data-ttu-id="1f0fd-119">Porque no hay ninguna coincidencia, el runtime produce una <xref:System.NullReferenceException> excepción.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-119">Because there is no match, the runtime throws a <xref:System.NullReferenceException> exception.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#4](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/nullreturn2.cs#4)]
     [!code-vb[System.NullReferenceException.Class#4](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/nullreturn2.vb#4)]  
  
     <span data-ttu-id="1f0fd-120">Para resolver este problema, pruebe el valor devuelto del método para garantizar que no se `null` antes de llamar a cualquiera de sus miembros, igual que en el ejemplo siguiente.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-120">To address this problem, test the method's return value to ensure that it is not `null` before calling any of its members, as the following example does.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#5](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/nullreturn2a.cs#5)]
     [!code-vb[System.NullReferenceException.Class#5](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/nullreturn2a.vb#5)]  
  
-   <span data-ttu-id="1f0fd-121">Está utilizando una expresión (por ejemplo, va a encadenar una lista de métodos o propiedades juntos) para recuperar un valor y, aunque está comprobando si el valor es `null`, el runtime produce aún un <xref:System.NullReferenceException> excepción.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-121">You're using an expression (for example, you're chaining a list of methods or properties together) to retrieve a value and, although you're checking whether the value is `null`, the runtime still throws a <xref:System.NullReferenceException> exception.</span></span> <span data-ttu-id="1f0fd-122">Esto se produce porque uno de los valores intermedios en la expresión devuelve `null`.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-122">This occurs because one of the intermediate values in the expression returns `null`.</span></span> <span data-ttu-id="1f0fd-123">Como resultado, el resto de las `null` nunca se evalúa.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-123">As a result, your rest for `null` is never evaluated.</span></span>  
  
     <span data-ttu-id="1f0fd-124">En el ejemplo siguiente se define un `Pages` objeto que almacena en caché información acerca de las páginas web, que se presentan por `Page` objetos.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-124">The following example defines a `Pages` object that  caches information about web pages, which are presented by `Page` objects.</span></span> <span data-ttu-id="1f0fd-125">El `Example.Main` método comprueba si la página web actual tiene un título no null y, si es así, muestra el título.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-125">The `Example.Main` method checks whether the current web page has a non-null title and, if it does, displays the title.</span></span> <span data-ttu-id="1f0fd-126">A pesar de esta comprobación, sin embargo, el método produce una <xref:System.NullReferenceException> excepción.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-126">Despite this check, however, the method throws a <xref:System.NullReferenceException> exception.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#6](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Chain1.cs#6)]
     [!code-vb[System.NullReferenceException.Class#6](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Chain1.vb#6)]  
  
     <span data-ttu-id="1f0fd-127">La excepción se produce porque `pages.CurrentPage` devuelve `null` si no hay información de página se almacena en la memoria caché.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-127">The exception is thrown because `pages.CurrentPage` returns `null` if no page information is  stored in the cache.</span></span> <span data-ttu-id="1f0fd-128">Esta excepción puede ser corregida por probar el valor de la `CurrentPage` propiedad antes de recuperar actual `Page` del objeto `Title` propiedad, como en el ejemplo siguiente se realiza:</span><span class="sxs-lookup"><span data-stu-id="1f0fd-128">This exception can be corrected by testing the value of the `CurrentPage` property before retrieving the current `Page` object's `Title` property, as the following example does:</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#7](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Chain2.cs#7)]
     [!code-vb[System.NullReferenceException.Class#7](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Chain2.vb#7)]  
  
-   <span data-ttu-id="1f0fd-129">Que está enumerando los elementos de una matriz que contiene los tipos de referencia y el intento de proceso de uno de los elementos produce un <xref:System.NullReferenceException> excepción.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-129">You're enumerating the elements of an array that contains reference types, and your attempt to process one of the elements throws a <xref:System.NullReferenceException> exception.</span></span>  
  
     <span data-ttu-id="1f0fd-130">En el ejemplo siguiente se define una matriz de cadenas.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-130">The following example defines a string array.</span></span> <span data-ttu-id="1f0fd-131">A `for` instrucción enumera los elementos de la matriz y llama a cada cadena <xref:System.String.Trim%2A> método antes de mostrar la cadena.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-131">A `for` statement enumerates the elements in the array and calls each string's <xref:System.String.Trim%2A> method before displaying the string.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#8](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Array1.cs#8)]
     [!code-vb[System.NullReferenceException.Class#8](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Array1.vb#8)]  
  
     <span data-ttu-id="1f0fd-132">Esta excepción se produce si supone que cada elemento de la matriz debe contener un valor distinto de null y el valor del elemento de matriz es en realidad `null`.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-132">This exception occurs if you assume that each element of the array must contain a non-null value, and the value of the array element is in fact `null`.</span></span> <span data-ttu-id="1f0fd-133">Se puede eliminar la excepción mediante la comprobación de si el elemento es `null` antes de realizar cualquier operación en ese elemento, como en el ejemplo siguiente se muestra.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-133">The exception can be eliminated by testing whether the element is `null` before performing any operation on that element, as the following example shows.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#9](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Array2.cs#9)]
     [!code-vb[System.NullReferenceException.Class#9](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Array2.vb#9)]  
  
-   <span data-ttu-id="1f0fd-134">A <xref:System.NullReferenceException> excepción se produce por un método que se pasa `null`.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-134">A <xref:System.NullReferenceException> exception is thrown by a method that is passed `null`.</span></span> <span data-ttu-id="1f0fd-135">Algunos métodos de validar los argumentos que se les pasan.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-135">Some methods validate the arguments that are passed to them.</span></span> <span data-ttu-id="1f0fd-136">Si lo hacen y uno de los argumentos es `null`, el método produce una <xref:System.ArgumentNullException?displayProperty=nameWithType> excepción.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-136">If they do and one of the arguments is `null`, the method throws an <xref:System.ArgumentNullException?displayProperty=nameWithType> exception.</span></span> <span data-ttu-id="1f0fd-137">De lo contrario, produce un <xref:System.NullReferenceException> excepción.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-137">Otherwise, it throws a <xref:System.NullReferenceException> exception.</span></span> <span data-ttu-id="1f0fd-138">En el ejemplo siguiente se muestra este escenario.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-138">The following example illustrates this scenario.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#3](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/example2.cs#3)]
     [!code-vb[System.NullReferenceException.Class#3](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/example2.vb#3)]  
  
     <span data-ttu-id="1f0fd-139">Para solucionar este problema, asegúrese de que el argumento pasado al método no es `null`, o controlar la excepción iniciada en un `try…catch…finally` bloque.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-139">To address this issue, make sure that the argument passed to the method is not `null`, or handle the thrown exception in a `try…catch…finally` block.</span></span> <span data-ttu-id="1f0fd-140">Para obtener más información, consulte [excepciones](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735).</span><span class="sxs-lookup"><span data-stu-id="1f0fd-140">For more information, see [Exceptions](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735).</span></span>  
  
 <span data-ttu-id="1f0fd-141">El siguiente producen instrucciones de lenguaje intermedio (MSIL) de Microsoft <xref:System.NullReferenceException>: `callvirt`, `cpblk`, `cpobj`, `initblk`, `ldelem.<type>`, `ldelema`, `ldfld`, `ldflda`, `ldind.<type>` , `ldlen`, `stelem.<type>`, `stfld`, `stind.<type>`, `throw`, y `unbox`.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-141">The following Microsoft intermediate language (MSIL) instructions throw <xref:System.NullReferenceException>: `callvirt`, `cpblk`, `cpobj`, `initblk`, `ldelem.<type>`, `ldelema`, `ldfld`, `ldflda`, `ldind.<type>`, `ldlen`, `stelem.<type>`, `stfld`, `stind.<type>`, `throw`, and `unbox`.</span></span>  
  
 <span data-ttu-id="1f0fd-142"><xref:System.NullReferenceException> utiliza HRESULT COR_E_NULLREFERENCE, que tiene el valor 0 x 80004003.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-142"><xref:System.NullReferenceException> uses the HRESULT COR_E_NULLREFERENCE, which has the value 0x80004003.</span></span>  
  
 <span data-ttu-id="1f0fd-143">Para obtener una lista de valores de propiedad iniciales para una instancia de <xref:System.NullReferenceException>, consulte el <xref:System.NullReferenceException.%23ctor%2A> constructores.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-143">For a list of initial property values for an instance of <xref:System.NullReferenceException>, see the <xref:System.NullReferenceException.%23ctor%2A> constructors.</span></span>  
  
 <span data-ttu-id="1f0fd-144">**Controlar excepciones NullReferenceException en código de versión**</span><span class="sxs-lookup"><span data-stu-id="1f0fd-144">**Handling NullReferenceException in release code**</span></span>  
  
 <span data-ttu-id="1f0fd-145">Es mejor evitar las excepciones NullReferenceException que controlar después de se produce.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-145">It's usually better to avoid a NullReferenceException than to handle it after it occurs.</span></span> <span data-ttu-id="1f0fd-146">Controlar una excepción puede dificultar el mantenimiento y la comprensión del código y, a veces, puede introducir otros errores.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-146">Handling an exception can make your code harder to maintain and understand, and can sometimes introduce other bugs.</span></span> <span data-ttu-id="1f0fd-147">Las excepciones NullReferenceException suelen ser errores sin recuperación.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-147">A NullReferenceException is often a non-recoverable error.</span></span> <span data-ttu-id="1f0fd-148">En esos casos, la mejor opción puede ser permitir que la excepción detenga la aplicación.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-148">In these cases, letting the exception stop the app might be the best alternative.</span></span>  
  
 <span data-ttu-id="1f0fd-149">Sin embargo, hay muchas situaciones en las que puede ser útil controlar el error:</span><span class="sxs-lookup"><span data-stu-id="1f0fd-149">However, there are many situations where handling the error can be useful:</span></span>  
  
-   <span data-ttu-id="1f0fd-150">La aplicación puede pasar por alto los objetos que son NULL.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-150">Your app can ignore objects that are null.</span></span> <span data-ttu-id="1f0fd-151">Por ejemplo, si la aplicación recupera y procesa los registros de una base de datos, puede caber la opción de pasar por alto cierto número de registros incorrectos que generan objetos NULL.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-151">For example, if your app retrieves and processes records in a database, you might be able to ignore some number of bad records that result in null objects.</span></span> <span data-ttu-id="1f0fd-152">Es posible que solo tenga que registrar los datos incorrectos en un archivo de registro o en la interfaz de usuario de la aplicación.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-152">Recording the bad data in a log file or in the application UI might be all you have to do.</span></span>  
  
-   <span data-ttu-id="1f0fd-153">La recuperación tras la excepción es posible.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-153">You can recover from the exception.</span></span> <span data-ttu-id="1f0fd-154">Por ejemplo, una llamada a un servicio web que devuelve un tipo de referencia podría devolver NULL si se interrumpe la conexión o se agota el tiempo de espera. Puede tratar de restablecer la conexión y volver a hacer la llamada.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-154">For example, a call to a web service that returns a reference type might return null if the connection is lost or the connection times out. You can attempt to reestablish the connection and try the call again.</span></span>  
  
-   <span data-ttu-id="1f0fd-155">Puede restaurar el estado de la aplicación a un estado válido.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-155">You can restore the state of your app to a valid state.</span></span> <span data-ttu-id="1f0fd-156">Por ejemplo, puede que esté realizando una tarea con varios pasos que exija guardar información en un almacén de datos antes de llamar a un método que lance una excepción NullReferenceException.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-156">For example, you might be performing a multi-step task that requires you to save information to a data store before you call a method that throws a NullReferenceException.</span></span> <span data-ttu-id="1f0fd-157">Si el objeto sin inicializar fuera a dañar el registro de datos, puede quitar los datos anteriores antes de cerrar la aplicación.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-157">If the uninitialized object would corrupt the data record, you can remove the previous data before you close the app.</span></span>  
  
-   <span data-ttu-id="1f0fd-158">La excepción se tiene que notificar.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-158">You want to report the exception.</span></span> <span data-ttu-id="1f0fd-159">Por ejemplo, si la causa del error fue un error del usuario de la aplicación, puede generar un mensaje para indicarle cómo proporcionar la información correcta.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-159">For example, if the error was caused by a mistake from the user of your app, you can generate a message to help him supply the correct information.</span></span> <span data-ttu-id="1f0fd-160">También puede registrar información sobre el error para facilitar la resolución del problema.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-160">You can also log information about the error to help you fix the problem.</span></span> <span data-ttu-id="1f0fd-161">Algunos marcos, como ASP.NET, tienen un controlador de excepciones de nivel superior que captura todos los errores para que la aplicación no se bloquee nunca. En ese caso, puede que la única forma de saber que se produce sea registrar la excepción.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-161">Some frameworks, like ASP.NET, have a high-level exception handler that captures all errors to that the app never crashes; in that case, logging the exception might be the only way you can know that it occurs.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="1f0fd-162">Inicializa una nueva instancia de la clase <see cref="T:System.NullReferenceException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-162">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NullReferenceException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NullReferenceException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NullReferenceException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="1f0fd-163">Inicializa una nueva instancia de la clase <see cref="T:System.NullReferenceException" />, estableciendo la propiedad <see cref="P:System.Exception.Message" /> de una nueva instancia en un mensaje proporcionado por el sistema que describe el error, como "Se ha encontrado el valor 'null' en el lugar donde era necesaria una instancia de un objeto".</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-163">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class, setting the <see cref="P:System.Exception.Message" /> property of the new instance to a system-supplied message that describes the error, such as "The value 'null' was found where an instance of an object was required."</span>
          </span>
          <span data-ttu-id="1f0fd-164">Este mensaje tiene en cuenta la referencia cultural del sistema actual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-164">This message takes into account the current system culture.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1f0fd-165">En la tabla siguiente se muestran los valores de propiedad iniciales para una instancia de <xref:System.NullReferenceException>.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-165">The following table shows the initial property values for an instance of <xref:System.NullReferenceException>.</span></span>  
  
|<span data-ttu-id="1f0fd-166">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-166">Property</span></span>|<span data-ttu-id="1f0fd-167">Valor</span><span class="sxs-lookup"><span data-stu-id="1f0fd-167">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="1f0fd-168">`null`.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-168">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="1f0fd-169">La cadena del mensaje de error localizado.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-169">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NullReferenceException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NullReferenceException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NullReferenceException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new NullReferenceException : string -&gt; NullReferenceException" Usage="new System.NullReferenceException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="1f0fd-170">Objeto <see cref="T:System.String" /> que describe el error.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-170">A <see cref="T:System.String" /> that describes the error.</span>
          </span>
          <span data-ttu-id="1f0fd-171">El contenido de <c>message</c> está diseñado para que sea inteligible.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-171">The content of <c>message</c> is intended to be understood by humans.</span>
          </span>
          <span data-ttu-id="1f0fd-172">El autor de llamada de este constructor debe asegurarse de que la cadena se haya adaptado para la referencia cultural actual del sistema.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-172">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="1f0fd-173">Inicializa una nueva instancia de la clase <see cref="T:System.NullReferenceException" /> con el mensaje de error especificado.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-173">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class with a specified error message.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1f0fd-174">En la tabla siguiente se muestran los valores de propiedad iniciales para una instancia de <xref:System.NullReferenceException>.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-174">The following table shows the initial property values for an instance of <xref:System.NullReferenceException>.</span></span>  
  
|<span data-ttu-id="1f0fd-175">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-175">Property</span></span>|<span data-ttu-id="1f0fd-176">Valor</span><span class="sxs-lookup"><span data-stu-id="1f0fd-176">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="1f0fd-177">`null`.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-177">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="1f0fd-178">Cadena con el mensaje de error.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-178">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected NullReferenceException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; NullReferenceException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new NullReferenceException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; NullReferenceException" Usage="new System.NullReferenceException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="1f0fd-179">Objeto que contiene los datos del objeto serializados.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-179">The object that holds the serialized object data.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="1f0fd-180">Información contextual acerca del origen o del destino.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-180">The contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="1f0fd-181">Inicializa una nueva instancia de la clase <see cref="T:System.NullReferenceException" /> con datos serializados.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-181">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class with serialized data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1f0fd-182">Durante la deserialización se llama a este constructor para reconstituir el objeto de excepción transmitido en una secuencia.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-182">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="1f0fd-183">Para obtener más información, consulte [serialización XML y SOAP](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="1f0fd-183">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NullReferenceException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NullReferenceException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NullReferenceException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new NullReferenceException : string * Exception -&gt; NullReferenceException" Usage="new System.NullReferenceException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="1f0fd-184">Mensaje de error que explica el motivo de la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-184">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="1f0fd-185">Excepción que es la causa de la excepción actual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-185">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="1f0fd-186">Si el parámetro <c>innerException</c> no es <see langword="null" />, se genera la excepción actual en un bloque <see langword="catch" /> que controla la excepción interna.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-186">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="1f0fd-187">Inicializa una nueva instancia de la clase <see cref="T:System.NullReferenceException" /> con el mensaje de error especificado y una referencia a la excepción interna que representa la causa de esta excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="1f0fd-187">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1f0fd-188">Una excepción que se produce como consecuencia directa de una excepción anterior debe incluir una referencia a esta última en la propiedad <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-188">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="1f0fd-189">La propiedad <xref:System.Exception.InnerException%2A> devuelve el mismo valor que se haya pasado al constructor, o bien `null` en caso de que la propiedad <xref:System.Exception.InnerException%2A> no le haya proporcionado al constructor el valor de la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-189">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="1f0fd-190">En la tabla siguiente se muestran los valores de propiedad iniciales para una instancia de <xref:System.NullReferenceException>.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-190">The following table shows the initial property values for an instance of <xref:System.NullReferenceException>.</span></span>  
  
|<span data-ttu-id="1f0fd-191">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-191">Property</span></span>|<span data-ttu-id="1f0fd-192">Valor</span><span class="sxs-lookup"><span data-stu-id="1f0fd-192">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="1f0fd-193">Referencia a la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-193">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="1f0fd-194">Cadena con el mensaje de error.</span><span class="sxs-lookup"><span data-stu-id="1f0fd-194">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
      </Docs>
    </Member>
  </Members>
</Type>